blog/_posts/2019-11-13-local-resource.md

```diff
@@ -0,0 +1,146 @@
+you'd rather compile your binary locally, and pull that compiled binary into your Docker image
+(or directly into your running container via Live Update).
+
+```python
+local_resource('compile-binary',
+    cmd='go build -i -o ./bin/the-binary github.com/my-org/my-app/',
+    deps='./my-app'
+)
+docker_build('gcr.io/my-org/my-app', context='./bin')
+```
+
+(Of course, the command you run doesn't have to be on that (only) affects your
+local filesystem. It might, for instance, be a script that is _invoked locally_
+but runs against your k8s cluster.)
+
+### Local Resource + Live Update = <3
+
+Local Resource and Live Update are two great tastes that taste great together.
+A useful pattern is to have a local resource which creates an artifact
+that is then picked up by a Live Update and synced to a running container.
+For example, here's how you might use this combination to generate protobufs:
+```python
+# whenever `helloworld.proto` changes, generate Python protobufs
+local_resource('proto',
+    cmd='protoc --proto_path=src --python_out=build/gen',
+    deps=['helloworld.proto']
+)
+
+# define your hello-world resource: k8s yaml + a Docker build
+k8s_yaml('hello-world.yaml')
+docker_build('hello-world', '.',
+    ignore='helloworld.proto',  # don't step on the local resource's toes
+    live_update=[ sync('.', '/app') ]
+)
+```
+
+Note that the `docker_build` call specifies `ignore='helloworld.proto'`. This is
+because we DON'T want an edit to that proto file to _directly_ kick off an update to
+our Docker image (in this case, a Live Update). Rather, an edit to `helloworld.proto`
+triggers the local resource to generate protobufs; when these protobufs appear on disk,
+they register as file changes and trigger a Live Update to the `hello-world` resource,
+i.e. they get `sync`'d to the container where `hello-world` is running.
+
+To see this pattern in action, check out [this example repo](https://github.com/windmilleng/local_resource_example).
+
+## Run occasional workflows (locally or against your cluster)
+
+There's another class of workflow that local resource can help with; commands that
+are a part of your development flow, but that you run only occasionally. This might
+be a task like "ensure GKE user 'foo' exists" or "refresh my credentials" or even
+"blow away my database".
+
+You can trigger a local resource at will with the "force update" button:
+
+!["force update" button](assets/img/force-update-button.png)
+
+By default, a local resource will run on startup. To disable this behavior, put the
+resource in `TRIGGER_MODE_MANUAL` and specify `auto_init=False`:
+```python
+local_resource('reset-db', cmd='reset_db.sh',
+    trigger_mode=TRIGGER_MODE_MANUAL, auto_init=False
+)
+```
+
+(`auto_init=False` is currently only compatible with `TRIGGER_MODE_MANUAL`. If
+you'd like a local resource that runs automatically in response to file changes
+but does NOT run on `tilt up`, [let us know](https://tilt.dev/contact).)
```

Comment:
throughout: we should use relative links instead of absolute
we should probably have a CI test for this